src/server/game/map/tile/city.ts

```diff
@@ -1,19 +1,32 @@
+import { Tile } from ".";
+import { Map } from "..";
+import { arrayIncludesCoords, getAdjacentCoords, getCoordInDirection, getDirection, getSmallesCoordsDiff, mod } from "../../../utils";
+import { World, Coords } from "../../world";
+import { ErrandType } from "./errand";
+import { Improvement } from "./improvement";
+import { Unit } from "./unit";
+
 export interface CityData {
   name: string;
-  civID: number;
+  civID?: number;
+  isBarbarian: boolean;
```

Comment:
Maybe instead of making civID nullable and having `isBarbarian`, barbarians could be a civ?
That might make things more complicated though...

Reply:
I did think of that, but it doesn't feel quite right (especially since, like I mentioned in an earlier comment, pretty much everything about a civ implies that it will be controlled by a player). If you have suggestions to the contrary, though, I'm very much interested to hear it 👀.

Reply:
Also, city-states? Does it make logical sense that they are civs too?

Reply:
I wouldn't _really_ distinguish between civs and city-states unless we have to.

I'm not sure about barbarians; it was just a thought.